backend/main/server/handlers/communities.go

```diff
@@ -0,0 +1,238 @@
+package handlers
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"github.com/DapperCollectives/CAST/backend/main/models"
+	s "github.com/DapperCollectives/CAST/backend/main/shared"
+	"github.com/go-playground/validator/v10"
+	"github.com/gorilla/mux"
+	"github.com/jackc/pgx/v4"
+	"github.com/rs/zerolog/log"
+)
+
+func AddCommunities(router *mux.Router, db *s.Database, ic *s.IpfsClient, fa *s.FlowAdapter, c s.Config) {
+	router.HandleFunc("/communities", getCommunities(db)).Methods("GET")
+	router.HandleFunc("/communities-for-homepage", getCommunitiesForHomePage(db)).Methods("GET")
+	router.HandleFunc("/communities/{id:[0-9]+}", getCommunity(db)).Methods("GET")
+	router.HandleFunc("/communities/{id:[0-9]+}", updateCommunity(db, fa, c)).Methods("PATCH", "OPTIONS")
+	router.HandleFunc("/communities", createCommunity(db, ic, fa, c)).Methods("POST", "OPTIONS")
+}
+
+func getCommunities(db *s.Database) func (w http.ResponseWriter, r *http.Request) {
+	return func (w http.ResponseWriter, r *http.Request) {
+		count, _ := strconv.Atoi(r.FormValue("count"))
+		start, _ := strconv.Atoi(r.FormValue("start"))
+	
+		if count > 25 || count < 1 {
+			count = 25
+		}
+		if start < 0 {
+			start = 0
+		}
+	
```

Comment:
this if block is repeating logic. We can put it into its own function

Reply:
Created helper function for this logic.